vscode/src/jsonrpc/agent-protocol.ts

```diff
@@ -542,6 +542,11 @@ export interface ProtocolTextDocument {
     /** @deprecated use `uri` instead. This property only exists for backwards compatibility during the migration period. */
     filePath?: string
     content?: string
+    /** `textDocument/didChange` fires `onDidChangeTextDocument` and we want to include `contentChanges` within.
+     * `contentChanges` can be determined based on the document event. The JetBrains client document event structure
+     * is different from `vscode.TextDocumentChangeEvent` - it misses `range`.
+     * We need derive the range based on the JetBrains event, and it's easier to do it on the agent side. */
+    textDocumentContentChangeEvents?: Omit<vscode.TextDocumentContentChangeEvent, 'range'>[]
```

Comment:
This is a mouthful but mirrors the existing pattern of prefixing VSC types with `Protocol`
```suggestion
    changes?: ProtocolTextDocumentContentChangeEvent[]
```